table/block.cc

```diff
@@ -155,32 +163,42 @@ void BlockIter::Seek(const Slice& target) {
   // Linear search (within restart block) for first key >= target
 
   while (true) {
-    if (!ParseNextKey() || Compare(key_.GetInternalKey(), target) >= 0) {
+    if (!ParseNextKey() ||
+        Compare(key_includes_seq_ ? key_.GetInternalKey() : key_.GetUserKey(),
+                seek_key) >= 0) {
```

Comment:
From code readability's point of the view, I prefer we do this branch inside Compare().